src/main/java/seedu/rc4hdb/logic/parser/commandparsers/ColumnManipulatorCommandParser.java

```diff
@@ -0,0 +1,125 @@
+package seedu.rc4hdb.logic.parser.commandparsers;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.rc4hdb.model.resident.fields.ResidentField.LETTER_TO_FIELD_NAME_MAPPINGS;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import seedu.rc4hdb.logic.commands.modelcommands.ColumnManipulatorCommand;
+import seedu.rc4hdb.logic.parser.Parser;
+import seedu.rc4hdb.logic.parser.exceptions.ParseException;
+
+/**
+ * Represents a parser for commands that can interact with Model to manipulate columns in the UI table.
+ */
+public abstract class ColumnManipulatorCommandParser implements Parser<ColumnManipulatorCommand> {
+
```

Comment:
Actually, `ListCommandParser`, `HideOnlyCommandParser` and `ShowOnlyCommandParser` could be done in this class. This will reduce the testing requirements of this section and reduce clutter in `ResidentBookParser`. Refer to the `FileCommandParser` to see what I mean.

Reply:
Interesting, thank you for the suggestion. I'll look into this. 😄